src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +261,427 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
+static ReduceNone data_assign;
+
+static inline void getCoordinate(VectorDims& coordinate, size_t offset, const VectorDims& shape) {
+    size_t shapeRank = shape.size();
+    for (int i = shapeRank - 1; i >= 0; i--) {
+        coordinate[i] = offset % shape[i];
+        offset /= shape[i];
+    }
+}
+
+namespace scatter_elements_update {
+struct TensorIterator {
+    TensorIterator(const VectorDims& squashed_shape, const int64_t squashed_axis) : m_squashed_shape(squashed_shape), m_squashed_axis(squashed_axis) {
+        OPENVINO_ASSERT(m_squashed_shape[m_squashed_axis] == 1);
+    }
+
+    std::array<size_t, 2> startover(const size_t start, const std::vector<size_t>& dataBlockND, const std::vector<size_t>& indicesBlockND) {
+        m_tensorIter.resize(m_squashed_shape.size(), 0);
+        getCoordinate(m_tensorIter, start, m_squashed_shape);
+
+        size_t i, dst_idx = 0, indices_idx = 0;
+        for (i = 0; i < static_cast<size_t>(m_squashed_axis); ++i) {
+            dst_idx += m_tensorIter[i] * dataBlockND[i + 1];
+            indices_idx += m_tensorIter[i] * indicesBlockND[i + 1];
+        }
+        for (i++; i < m_squashed_shape.size(); ++i) {
+            dst_idx += m_tensorIter[i] * dataBlockND[i + 1];
+            indices_idx += m_tensorIter[i] * indicesBlockND[i + 1];
+        }
+
+        return {dst_idx, indices_idx};
+    }
+
+    void increment(std::array<size_t, 2>& offsets, const std::vector<size_t>& dataBlockND, const std::vector<size_t>& indicesBlockND) {
+        for (int64_t j = m_squashed_shape.size() - 1; j >= 0; j--) {
+            m_tensorIter[j]++;
+            if (m_tensorIter[j] < m_squashed_shape[j]) { // no need check if (j != axis) as it is squashed
+                offsets[0] += dataBlockND[j + 1];
+                offsets[1] += indicesBlockND[j + 1];
+                break;
+            } else {
+                m_tensorIter[j] = 0;
+                size_t i = 0;
+                for (offsets[0] = 0, offsets[1] = 0; i < m_squashed_axis; ++i) {
+                    offsets[0] += m_tensorIter[i] * dataBlockND[i + 1];
+                    offsets[1] += m_tensorIter[i] * indicesBlockND[i + 1];
+                }
+                for (i++; i < m_squashed_shape.size(); ++i) {
+                    offsets[0] += m_tensorIter[i] * dataBlockND[i + 1];
+                    offsets[1] += m_tensorIter[i] * indicesBlockND[i + 1];
+                }
+            }
+        }
+    }
+
+    VectorDims m_tensorIter;
+    const VectorDims m_squashed_shape;
+    const size_t m_squashed_axis;
+};
+};   // namespace scatter_elements_update
+
+// output[indices[i][j][k]][j][k] = updates[i][j][k] if axis = 0,
+// output[i][indices[i][j][k]][k] = updates[i][j][k] if axis = 1,
+// output[i][j][indices[i][j][k]] = updates[i][j][k] if axis = 2.
+template <typename DataType, typename func_t>
+void ScatterUpdate::scatterElementsUpdate(const MemoryPtr& mem_data, const MemoryPtr& mem_indices, const MemoryPtr& mem_updates,
```

Comment:
Have you compared performance of new impl with the old one for cases reduction_type = ov::op::v12::ScatterElementsUpdate::Reduction::NONE? I am wondering if we sure there is no perf regression

Reply:
yes, there is no perf regression. Please check the ticket comments, I have perf regression tests with models. These models in model cache are using Reduction::None (SEU-3).

Reply:
yes, confirmed with op level perf regression test. Please check comments in corresponding ticket CVS-129813.